TESTS/mbed_platform/error_handling/main.cpp

```diff
@@ -0,0 +1,507 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "greentea-client/test_env.h"
+#include "utest/utest.h"
+#include "unity/unity.h"
+#include "mbed.h"
+#include <LittleFileSystem.h>
+#include "HeapBlockDevice.h"
+
+using utest::v1::Case;
+
+/** Test logging of system errors
+ *  and ensure the status/erro code is correct
+ */
+void test_system_errors()
+{
+    MbedErrorStatus error = MAKE_ERROR(MODULE_APPLICATION, ERROR_CODE_UNKNOWN);
+    SET_WARNING(error, "Error Unknown", 0xAABBCCDD );
```

Comment:
Why do we have SET_WARNING as an API? how is this different then tracing feature MBED_WARN or tr_warn?

Reply:
SET_WARNING captures error context, can call application hooks, stores the error error context etc but tracing does not(or rather should not). Error handling should be seen differently from tracing and both are needed.

Reply:
Please see my reply above, unfortunately we cannot use mbed_trace APIs as of now due to its extensive use of resources. Note that this needs to work with handling of exceptions as well.